pytorch_lightning/utilities/types.py

```diff
@@ -43,7 +44,75 @@
     Dict[str, Sequence[DataLoader]],
 ]
 EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
+
+
+# Copied from `torch.optim.lr_scheduler.pyi`
+# Missing attributes were added to improve typing
+class _LRScheduler:
+    optimizer: Optimizer
+
+    def __init__(self, optimizer: Optimizer, last_epoch: int = ...) -> None:
+        ...
+
+    def state_dict(self) -> dict:
+        ...
+
+    def load_state_dict(self, state_dict: dict) -> None:
+        ...
+
+    def get_last_lr(self) -> List[float]:
+        ...
+
+    def get_lr(self) -> float:
+        ...
+
+    def step(self, epoch: Optional[int] = ...) -> None:
+        ...
+
+
+# Copied from `torch.optim.lr_scheduler.pyi`
+# Missing attributes were added to improve typing
+class ReduceLROnPlateau:
+    in_cooldown: bool
+    optimizer: Optimizer
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        mode: str = ...,
+        factor: float = ...,
+        patience: int = ...,
+        verbose: bool = ...,
+        threshold: float = ...,
+        threshold_mode: str = ...,
+        cooldown: int = ...,
+        min_lr: float = ...,
+        eps: float = ...,
+    ) -> None:
+        ...
+
+    def step(self, metrics: Any, epoch: Optional[int] = ...) -> None:
+        ...
+
+    def state_dict(self) -> dict:
+        ...
+
+    def load_state_dict(self, state_dict: dict) -> None:
+        ...
+
+
 # todo: improve LRSchedulerType naming/typing
-LRSchedulerTypeTuple = (_LRScheduler, ReduceLROnPlateau)
-LRSchedulerTypeUnion = Union[_LRScheduler, ReduceLROnPlateau]
-LRSchedulerType = Union[Type[_LRScheduler], Type[ReduceLROnPlateau]]
+LRSchedulerTypeTuple = (torch.optim.lr_scheduler._LRScheduler, torch.optim.lr_scheduler.ReduceLROnPlateau)
+LRSchedulerTypeUnion = Union[torch.optim.lr_scheduler._LRScheduler, torch.optim.lr_scheduler.ReduceLROnPlateau]
+LRSchedulerType = Union[Type[torch.optim.lr_scheduler._LRScheduler], Type[torch.optim.lr_scheduler.ReduceLROnPlateau]]
+
+
+class LRSchedulerConfig(TypedDict):
```

Comment:
Since we have this now, we can remove `_get_default_scheduler_config`

Reply:
This would be a BC for Flash.

Reply:
@carmocca could you elaborate how it is now possible to remove it?

Reply:
We cannot set defaults for the `TypedDict`.

Ideally, this would be a dataclass, which then is used in our code. then we don't need the function to create the dict

Reply:
Right now the code expects a dict everywhere. I think it would be best to leave this for a separate PR in which this can be updated across the code base if needed and provide a way to make it work with flash

Reply:
will look into it #11211